musicbrainz/csv/convert_to_csv.py

```diff
@@ -105,6 +112,16 @@ def extract(data, value: dict, first_level: bool = True, key: str = ""):
         return
     
 def convert_dict_to_csv(dictionary_list, filename):
+    """
+    (list, str) -> None
```

Comment:
Also, just a matter of "thinking about code":  a function that returns `None` is a red flag. That means that it's got "side effects" and is modifying data outside of its inner scope. For classes this is kinda expected, since the methods on a class can interact with class and instance variables. But for standalone functions, this indicates that your function is changing something somewhere else. 

This, of course, is one of the main source of bugs in code: Variables being modified in many places, and no "ownership" of the data at any given point in time. So you may find your data being in an unexpected state because the application of the processing on that data may not be in an order that you expect it in. 

The TL;DR is that, unless you're writing code, your functions should always return a value, and given the same arguments, the return value should be the same. This will help you avoid bugs you didn't even know you could have!

Reply:
It is not modifying outside variables, it is a serialization process of writing data into a file.

Reply:
If it’s returning None then that is not possible. You are modifying a single instance of a dict which is declared outside of the scope of this function.

Reply:
I am only reading from that dict, how am I modifying it?

Reply:
If the return value of your function is None, the. It must be modifying data somewhere, right? Otherwise there is no effect of your function….

Reply:
Sorry, I just noticed that this was attached to the wrong function. I was thinking about the `extract` method above, but I notice that my comment is on this function. (It's the same observation, just on a different method).

Reply:
Yes, the extract function is modifying a dictionary defined outside.

Reply:
You can't. Not easily, anyway. In a purely functional language, you would define your recursion to return the dictionary at each step, which would then get modified in the next step, and returned, until it ended. That's not how you have designed yours. 

What I haven't mentioned (up to this point) is that using the `global` keyword is generally frowned upon in Python, for exactly this same reason. It makes for code that is difficult to debug, because the value can be changed in many places. 

No action needed for this (you've had enough changes on this file) but when you write your next converter, try not to use `global`...